### PR TITLE
Fixing a bug where the optional prefix is being included even when a …

### PR DESCRIPTION
**Pull request by scottfrazer**
…value is not specified

### CODE REVIEW

File: src/main/scala/cromwell/binding/command/ParameterCommandPart.scala [hunk 1]
```diff
@@ -90,6 +90,7 @@ case class ParameterCommandPart(wdlType: WdlType, name: String,
      * turned into strings, but a more sophisticated solution will be
      * needed for compound types */
     paramValueEvaluated match {
+      case str:WdlString if str.value.isEmpty => ""
```
Comment (geoffjentry) [inline src/main/scala/cromwell/binding/command/ParameterCommandPart.scala]: Curiousity Q -> is the `isEmpty` also guarding against `null` (as in is there an expectation that it could actually be null) and if so can we root it out at the java/scala border?

Along those lines if null doesn't enter into it, something like `case WdlString("") => ""` should work and manage to be both more clear & terse at the same time. But if `null` is a concern that's clearly not so good :)

**Comment by geoffjentry** [comment]
:+1:

File: src/main/scala/cromwell/binding/command/ParameterCommandPart.scala (hunk 1 above)
Comment (scottfrazer) [inline src/main/scala/cromwell/binding/command/ParameterCommandPart.scala]: No, this doesn't have anything to do with null values.  I can try your suggestion, that looks cleaner.  This is basically to short circuit so we don't prepend the prefix if an empty value was specified for a parameter.

**Comment by scottfrazer** [comment]
Since you already reviewed: @geoffjentry is the first reviewer

File: src/test/scala/cromwell/OptionalParamWorkflowSpec.scala [hunk 2]
```diff
@@ -35,4 +37,35 @@ class OptionalParamWorkflowSpec extends CromwellTestkitSpec("OptionalParamWorkfl
...
+         |  command {
+         |    find ${File root} ${"-name " pattern?}
+         |  }
+         |}
+         |
+         |workflow wf {
+         |  call find
+         |}
+       """.stripMargin
+      val ns = WdlNamespace.load(wf, BackendType.LOCAL)
```
Comment (kshakir) [inline src/test/scala/cromwell/OptionalParamWorkflowSpec.scala]: For tests that are expected to pass, the loaded `wf: WdlSource` should be in `SampleWdl`, not inline above. I'm not sure about tests that are expected to fail, like this one. So, unless someone else can chime in with something better, I would leave it as is.

TL;DR: This is ok.

Comment (scottfrazer) [inline src/test/scala/cromwell/OptionalParamWorkflowSpec.scala]: Well I'm not opposed to moving it there, though usually I put stuff in `SampleWdl` when I intend to run it through the actor system, but not necessarily for just parsing.  SampleWdl makes you define things like `RawInputs` which felt awkward for me considering I wasn't going to use them.  I know I've done the inlining in other places in the past and it works well especially for very targeted syntax checks.

File: src/test/scala/cromwell/OptionalParamWorkflowSpec.scala [hunk 3]
```diff
@@ -35,4 +37,35 @@ class OptionalParamWorkflowSpec extends CromwellTestkitSpec("OptionalParamWorkfl
...
+      val findTask = ns.findTask("find").getOrElse {
```
Comment (kshakir) [inline src/test/scala/cromwell/OptionalParamWorkflowSpec.scala]: Quickly scanning, I see three dot-infix `.xyz{` like this one, that should be converted to either `.xyz(` or ` xzy {`.

**Comment by kshakir** [comment]
Minor infix patches, and then unless anyone else chimes in, :+1: for merge.
